### ISSUE TITLE
DOC: Remove versionadded/versionchanged up to 1.5.0

### ISSUE THREAD

**Issue by phofl** [pandas-dev/pandas#51071]
### Pandas version checks

- [X] I have checked that the issue still exists on the latest versions of the docs on `main` [here](https://pandas.pydata.org/docs/dev/)

### Location of the documentation

We have many versionadded/versionchanged notes in our API docs. This clutters our docstrings quite a bit, see https://pandas.pydata.org/docs/dev/reference/api/pandas.read_csv.html for example and is actually not really relevant for most users. I'd suggest removing everything that was added up to 1.4.5, e.g. 1.4.5 or lower to clean this up a bit.

We have the version switcher if people want to go back. Also, 2.0 is a major release so I think it's a good opportunity to clean this up.

### Documentation problem

See above

### Suggested fix for documentation

Remove the notes

**Comment by phofl** [comment]
I'd at least remove up to 1.3.5, 1.4.x and 1.5.x will cover over a year by the time 2.0 comes out, this should be sufficient. I mean it's still there if you switch to the 1.5.x docs

**Comment by rhshadrach** [comment]
> I'd at least remove up to 1.3.5, 1.4.x and 1.5.x will cover over a year by the time 2.0 comes out, this should be sufficient.

I don't agree here, even the latest LTS runtime environment from Databricks is using pandas 1.3.5. In my experience, many users install packages and don't change their versions unless something breaks. This is bad practice in my opinion, but I think  it leaves us with users making use of older versions regardless.

**Comment by phofl** [comment]
I agree, but you could make this argument for every version we ever released. Do you think someone would upgrade from 1.0 to 2.0 straight? I would expect users to upgrade maybe 2 minor versions at a time but no more. 

In general it's a bad idea to upgrade from anything but 1.5 to 2.0, no chance of seeing all deprecation warnings before.

**Comment by rhshadrach** [comment]
> I would expect users to upgrade maybe 2 minor versions at a time but no more.

I still see frequent use of 1.3.x. I can't be certain, but I would guess that a good portion of users that look at the documentation do so without switching to the appropriate version; the default page that loads is the most recent version. Where I foresee a bad user experience is to have users on 1.3.x looking at documentation of the most recent version and seeing some argument exists or behaves a certain way that is not the case in their version. By having the versionchanged et al, it would give a way for them to realize what's going on.

In other words, I don't think the number of versions that get upgraded is the thing that's important; just what versions are in common use currently.

**Comment by lithomas1** [comment]
I can pull download counts for various pandas version to see which versions are still in use from the GBQ PyPI dataset.
(Had this working before, but my heroku dyno got shut down)

**Comment by lithomas1** [comment]
Currently the top 10 versions I have are (for downloads today, but the monthly counts are similar):

1.5.3: 18.371%
1.3.5: 15.966%
1.1.5: 11.428%
1.2.5: 5.278%
0.24.2: 4.853%
1.0.5: 3.793%
1.3.4: 3.179%
0.25.3: 2.793%
1.2.1: 2.339%
1.0.1: 2.172%
Other: 29.828%

![bokeh_plot](https://user-images.githubusercontent.com/47963215/216774936-3d779630-ff0a-42c8-86b8-08169399d820.png)

**Comment by lithomas1** [comment]
Going by this, I would be +1 on removing up to 1.1.0 and -1 on everything else.

IMO, we should try to set a threshold for things (docs versionadded/versionchanged, platform support too) like this. I can write up another PDEP if folks are interested.

**Comment by rhshadrach** [comment]
Agreed. For a PDEP, I would suggest using a constant # of previous versions rather than basing it on usage stats.

**Comment by jorisvandenbossche** [comment]
FWIW given this only about docs and not actual runtime, I don't think that's worth a PDEP. Just discussing this and writing up that as a guideline in our contributing docs can be sufficient I think.

(it's something else for how long runtime dependencies are supported, like NEP29)

**Comment by lithomas1** [comment]
Yeah, I would like to formalize things like support for new architectures, version bumps of deps, and dropping architectures as well.

Right now there's ambiguity over things like if a pyarrow version is too new to set as the minimum, how do determine threshold for adding platforms(e.g. Alpine is under consideration right now), and whether we can remove things like 32-bit support.

For now, how about lets assume people are currently using the last four pandas versions (based on usage details, which may or may not be accurate due to noise from things like CI puling pandas).

This would mean we drop versionchanged in docs up to 1.3.5.

**Comment by jbrockmendel** [comment]
closed by #63035 and others
